pkg/tenant/topics.go

```diff
@@ -31,18 +35,123 @@ func (s *Server) TopicList(c *gin.Context) {
 	c.JSON(http.StatusNotImplemented, "not implemented yet")
 }
 
-func (s *Server) TopicDetail(c *gin.Context) {
+func (s *Server) TopicCreate(c *gin.Context) {
 	c.JSON(http.StatusNotImplemented, "not implemented yet")
 }
 
-func (s *Server) TopicCreate(c *gin.Context) {
-	c.JSON(http.StatusNotImplemented, "not implemented yet")
+// TopicDetail retrieves a summary detail of a topic with a given ID
+// and returns a 200 OK response.
+func (s *Server) TopicDetail(c *gin.Context) {
+	var (
+		err   error
+		reply *api.Topic
+	)
+
+	// Get the topic ID from the URL and return a 400 response
+	// if the topic does not exist.
+	var topicID ulid.ULID
+	if topicID, err = ulid.Parse(c.Param("topicID")); err != nil {
+		log.Error().Err(err).Msg("could not parse topic ulid")
+		c.JSON(http.StatusBadRequest, api.ErrorResponse("could not parse topic id"))
+		return
+	}
+
+	// Get the specified topic from the database and return a 404 response
+	// if it cannot be retrieved.
+	//
+	// Route: /topic/:topicID
+	var topic *db.Topic
+	if topic, err = db.RetrieveTopic(c.Request.Context(), topicID); err != nil {
+		log.Error().Err(err).Str("topicID", topicID.String()).Msg("could not retrieve topic")
+		c.JSON(http.StatusNotFound, api.ErrorResponse("could not retrieve topic"))
+		return
+	}
+
+	reply = &api.Topic{
+		ID:   topic.ID.String(),
+		Name: topic.Name,
+	}
+
+	c.JSON(http.StatusOK, reply)
 }
 
+// TopicUpdate updates the record of a topic with a given ID and
+// returns a 200 OK response.
+//
+// Route: /topic/:topicID
 func (s *Server) TopicUpdate(c *gin.Context) {
-	c.JSON(http.StatusNotImplemented, "not implemented yet")
+	var (
+		err   error
+		topic *api.Topic
+	)
+
+	// Get the topic ID from the URL and return a 400 response if
+	// the topic ID is not a ULID.
+	var topicID ulid.ULID
+	if topicID, err = ulid.Parse(c.Param("topicID")); err != nil {
+		log.Error().Err(err).Msg("could not parse topic ulid")
```

Comment:
should this be `could not parse topic id`?

Reply:
The current message felt a little more descriptive since the topicID is a ulid. I'll update the message in the line below to match this message.